subprojects/core/src/main/java/org/gradle/api/internal/TaskOutputsInternal.java

```diff
@@ -37,19 +37,12 @@
     void setHistory(TaskExecutionHistory history);
 
     /**
-     * Check if caching is explicitly enabled for the task outputs.
+     * Yields information about the cacheability of the outputs.
      */
-    boolean isCacheEnabled();
+    TaskCaching getTaskCaching();
```

Comment:
I would probably call the type `TaskOutputCaching` and the method `getCaching()` as `Task` and `Output` is already in the name of the type the method is on.